### PR TITLE
ci: add unittest to ci; add TODOs for fail testing

### PR DESCRIPTION
**Pull request by sljeff**
- Add local config and docker-compose.yml for local testing
- Add CI unittest
- Currently there are some unittests can not pass on my local env; TODOs added
- Why remove all `t.Parallel()`?
    - some map init will fail (data race detected)
    - data in db will be inconsistent (when updating and checking concurrently)

### CODE REVIEW

**Comment by sljeff** [comment]
image

@Songkeys @lazzyRabbit 

Added unittest CI

Although tens of files have been updated, all of them are testing/ci/script files. You may have a quick scan and merge it.

File: hub/database/datatype/attachment_test.go [hunk 1]
```diff
@@ -23,8 +23,6 @@ func init() {
 }
 
 func TestAttachment(t *testing.T) {
-	t.Parallel()
```
Comment (Songkeys) [inline hub/database/datatype/attachment_test.go:26]: why remove this?

Comment (sljeff) [reply hub/database/datatype/attachment_test.go:26]: look the 1st floor `Conversation`

Comment (Songkeys) [reply hub/database/datatype/attachment_test.go:26]: i see.

**Pull request merged**